packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -189,12 +187,30 @@ const ImageDialog = ({ handleClose }) => {
   const MediaLibraryDialog = components['media-library'];
 
   const insertImages = (images) => {
-    // Image node created using select or existing selection node needs to be deleted before adding new image nodes
+    // If the selection is inside a list, split the list so that the modified block is outside of it
+    Transforms.unwrapNodes(editor, {
+      match: (node) => node.type === 'list',
+      split: true,
+    });
+
+    // Save the path of the node that is being replaced by an image to later insert the images there
+    // It's the closest full block node above the selection
+    const [, pathToInsert] = Editor.above(editor, {
+      match(node) {
+        const isInlineNode = ['text', 'link'].includes(node.type);
```

Comment:
why do we need this check? what happens if we do not check this

Reply:
If your cursor was inside a link for example, the pathToInsert would be wrong, it would create the image inside the list again